compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaLexer.java

```diff
@@ -494,24 +499,31 @@ private STToken processEqualOperator() {
     private STToken processNumericLiteral(int startChar) {
         int nextChar = peek();
         if (isHexIndicator(startChar, nextChar)) {
-            return processHexIntLiteral();
+            return processHexLiteral();
         }
 
         int len = 1;
         while (true) {
             switch (nextChar) {
-                case '.':
+                case LexerTerminals.DOT:
                 case 'e':
                 case 'E':
+                case 'f':
+                case 'F':
+                case 'd':
+                case 'D':
                     // In sem-var mode, only decimal integer literals are supported
                     if (this.mode == ParserMode.IMPORT) {
                         break;
                     }
 
-                    // TODO: handle float
-                    reader.advance();
-                    processInvalidToken();
-                    return readToken();
+                    // Integer part of the float cannot have a leading zero
+                    if (startChar == '0' && len > 1) {
+                        processInvalidToken();
+                        return readToken();
+                    }
+
+                    return processDecimalFloatLiteral(false);
```

Comment:
Add a comment saying that we wouldn't be reaching here, if the floating-point starts with a dot.

Otherwise, anyone who will check this code later, might think this was a mistake :)